dsm/data_source_sobject.go

```diff
@@ -99,19 +99,37 @@ func dataSourceSobject() *schema.Resource {
 
 func dataSourceSobjectRead(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
 	var diags diag.Diagnostics
+	var req map[string]interface{}
+	var err diag.Diagnostics
 
 	security_object := map[string]interface{}{
 		"name": d.Get("name").(string),
 	}
 
-	req, err := m.(*api_client).APICallBody("POST", "crypto/v1/keys/export", security_object)
-	if err != nil {
-		diags = append(diags, diag.Diagnostic{
-			Severity: diag.Error,
-			Summary:  "[DSM SDK] Unable to call DSM provider API client",
-			Detail:   fmt.Sprintf("[E]: API: POST crypto/v1/keys/export: %v", err),
-		})
-		return diags
+	if d.Get("export").(bool) {
+		req, err = m.(*api_client).APICallBody("POST", "crypto/v1/keys/export", security_object)
+		if err != nil {
+			diags = append(diags, diag.Diagnostic{
+				Severity: diag.Error,
+				Summary:  "[DSM SDK] Unable to call DSM provider API client",
+				Detail:   fmt.Sprintf("[E]: API: POST crypto/v1/keys/export: %v", err),
+			})
+			return diags
+		}
+	} else {
+		var reqList []interface{}
+		reqList, err = m.(*api_client).APICallList("GET", "crypto/v1/keys?name=" + d.Get("name").(string))
+		if err == nil && len(reqList) > 0 {
+			req = reqList[0].(map[string]interface{})
+		}
+		if err != nil {
+			diags = append(diags, diag.Diagnostic{
+				Severity: diag.Error,
+				Summary:  "[DSM SDK] Unable to call DSM provider API client",
+				Detail:   fmt.Sprintf("[E]: API: POST crypto/v1/keys?name=%s: %v", d.Get("name").(string), err),
```

Comment:
I think the `Detail` message is incorrect. We are calling `GET crypto/v1/keys?name=` and the error says `POST crypto/v1/keys?name=`.

Reply:
Thanks for pointing that out. Will correct it.